eng/build.sh

```diff
@@ -337,6 +337,6 @@ if [[ "$test_core_clr" == true ]]; then
   if [[ "$ci" != true ]]; then
     runtests_args="$runtests_args --html"
   fi
-  dotnet exec "$scriptroot/../artifacts/bin/RunTests/${configuration}/net7.0/RunTests.dll" --tfm net7.0 --configuration ${configuration} --logs ${log_dir} --dotnet ${_InitializeDotNetCli}/dotnet $runtests_args
```

Comment:
So not only was this change fragile, it was broken. Compiler hasn't been running our unit tests that targeted `net8.0` on Unix. And yes there are failures 😢

Reply:
Will use https://github.com/dotnet/roslyn/issues/71571 to track the failing tests